### DISCUSSION TITLE
Override default error generic type typescript-react-query

### DISCUSSION THREAD

**Discussion by gmencz** [dotansimha/graphql-code-generator#5377] [Q&A] [answered]
Hi, currently this is the typescript being generated:
```ts
export const useMostRecentUsersQuery = <
      TData = MostRecentUsersQuery,
      TError = unknown
    >(
      variables: MostRecentUsersQueryVariables, 
      options?: UseQueryOptions<MostRecentUsersQuery, TError, TData>
    ) => 
    useQuery<MostRecentUsersQuery, TError, TData>(
      ['MostRecentUsers', variables],
      fetcher<MostRecentUsersQuery, MostRecentUsersQueryVariables>(MostRecentUsersDocument, variables),
      options
    );
```

However, I know that all of my errors will be of a custom type we have so it would be nice if instead of generating `TError = unknown` we could override that with `TError = OurCustomErrorType` so that we don't have to specify that same error type as a generic on every react-query operation, is this currently possible?

**Comment by dotansimha** [comment, marked as answer]
GraphQL spec doesn't have any way to define types of errors, and usually it's done based on project needs. We can't really expect specific errors, or even error types, so all we can do here is to allow you to override it as a generic. 

I added an option to override the default error type, if that helps: https://github.com/dotansimha/graphql-code-generator/pull/5447 . I will be available in the next release (`1.1.0`, and already available in alpha: `@graphql-codegen/typescript-react-query@1.1.0-alpha-639d7dc8.0`)

**Reply by gmencz** [reply]
Awesome, that's exactly what I need.
